lib/spack/spack/test/llnl/util/filesystem.py

```diff
@@ -712,6 +712,9 @@ def test_is_nonsymlink_exe_with_shebang(tmpdir):
         assert not fs.is_nonsymlink_exe_with_shebang("symlink_to_executable_script")
 
 
+@pytest.mark.skipif(
+    sys.platform == "win32" and not _windows_can_symlink(), reason="Requires elevated privileges."
+)
```

Comment:
FWIW GHA generally guarantees developer permissions are enabled. If you'd like I can track down that blog post when I get a second.
Nit: the reason is that developer mode needs to be on. That does itself require one off admin privileges, but just looking at this reason, I would think elevating my command prompt would solve the issue, which, IIRC it does not unless dev mode is enabled. Additionally with this reason message, if this test was running without admin privs (but dev mode is on) users may get confused.

Reply:
@scheibelp to be clear, there is a test for unix & windows w/admin, and there is a test for just windows without admin. One test or the other will run no matter where you are. The only difference is the expected answer if the link is not a symlink, so there isn't a failure when running unit tests without admin. 

@johnwparent I will update the reason to say "Requires developer mode enabled"

Reply:
Thanks!

Reply:
I agree we need to be able to skip this when the user runs `spack unit-test` on a system without developer mode enabled. However, I want a way to ensure that this test runs in our CI (i.e. if we accidentally disable developer mode for our runners, I don't want this test to start automatically skipping in CI). 

One way to do that is to add a config variable like `config:require_developer_mode`: if that is set (to True) then we could fail (and then the CI configuration could set that on).

Reply:
> is the concern that we have a disabled dev mode on the CI machine and that skipped test allows this behavior to fail? 

yes

> IMO that'd be better handled from a system check sort of test that would fail independently of this one.

that sounds good to me. Using config was my first thought about how to enforce this when running CI, but IIRC you can also specify additional CI checks in `ci.yml`